src/modules/achievements/sagas.js

```diff
@@ -0,0 +1,135 @@
+import { all, call, fork, put, select, take, takeLatest } from 'redux-saga/effects'
+import ownT from './types'
+import ownA from './actions'
+import api from 'services/api'
+import stream from 'services/stream'
+import S from 'modules/selectors'
+import T from 'modules/types'
+import { eventChannel } from 'redux-saga'
+import blockchains from 'configurables/blockchains'
+
+const create = function * ({ link, title }) {
+  try {
+    const primaryBlockchainKey = blockchains.primary.key
+    const userAddress = yield select(S.wallets.primaryAddress)
+    yield call(api.createAchievement(primaryBlockchainKey), {
```

Comment:
Here we pass: address, link, name, title, token, user.
Getstream backend service accepts: userAddress, link, title.
I suggest to rename address to userAddress, and remove redundant fields that can be fetched from stream by address (user, name)
We can keep token for now to compare it with token saved by user and check authorization on service, but later token also will not be used for authorization, but instead blockchain will verify signature of address.

Reply:
I :
- changed address for userAddress everywhere.
- removed userName everywhere
- removed token (we do no `checkAddressMatchUser` anymore)

Note that in the end I do not intend to deploy "Getstream backend service" but instead just implement some changes in your existing backend.
Because at first I thought it was lots of calls but in the end for this first version it's only 4 calls which already match existing calls. So there is no sense into adding a service for this.